### PR TITLE
GROK-13229: SA: Variance-based sensitivity analysis

### PR DESCRIPTION
**Pull request by vmakarichev**
Sensitivity Analysis computations are implemented. Further, their megration to compute-utils is planned.

### CODE REVIEW

File: packages/SA/src/inputTools.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+  for (let i = 0; i < length; ++i)
+    columnSource[i] = numInputInfo.min + step * randomData[i];
+
+  numInputInfo.column = column;
+  
+  return column;
+}
+
+export function createVariedNumericalInputColumns(samplesCount: number,
```
Comment (AufarZakiev) [inline packages/SA/src/inputTools.ts:41]: It is hard to debug functions when they vary input params. Is it possible to rewrite this function without changing inputs params with no performance penalty?

File: packages/SA/src/inputTools.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+import {getGeneratedColumnsData} from './inputsGeneration';
+
+export type VariedNumericalInputInfo = {
+  name: string,
+  caption: string | undefined,
+  type: DG.COLUMN_TYPE,
+  min: number,
+  max: number,
+  column: DG.Column | undefined
```
Comment (AufarZakiev) [inline packages/SA/src/inputTools.ts:15]: This structure could be simpler. 
`name`, `caption`, `type` could be stored directly in DG.Property

`column` field is always undefined before values generation. It is easier to create separate structure for filled info.
For example,
```
type VariedNumericalInputInfo = {
  prop: DG.Property,
  min: number,
  max: number,
}
```
and 
```
type VariedNumericalInputValues = VariedNumericalInputInfo & {column: DG.Column}
```

What do you think about it?

File: packages/SA/src/outputTools.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+type OutputInfo = {
+  name: string,  
+  type: DG.TYPE
+};
+
+function getOutputsInfo(func: DG.Func): OutputInfo[] {
+  const ouputInfo = Array<OutputInfo>(0);
+
+  for (const item of func.outputs)
```
Comment (AufarZakiev) [inline packages/SA/src/outputTools.ts:17]: Could be replaced in one-line by `map` function

**Review by AufarZakiev — CHANGES_REQUESTED** [review]

File: packages/SA/src/sensitivityAnalysis.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    for (const funcCall of this.funcCalls)
+      await funcCall.call();
+  }
+
+  async perform(): Promise<DG.DataFrame> {
+    await this.run();
+
+    const columns: DG.Column[] = [];
+
+    for (const input of this.variedInputs)
```
[the remark this answers is not in the document]
Comment (vmakarichev) [reply packages/SA/src/sensitivityAnalysis.ts:59]: Done

File: packages/SA/src/inputTools.ts (hunk 1 above)
Comment (vmakarichev) [reply packages/SA/src/inputTools.ts:41]: Done

File: packages/SA/src/outputTools.ts (hunk 3 above)
Comment (vmakarichev) [reply packages/SA/src/outputTools.ts:17]: Done

File: packages/SA/src/inputTools.ts (hunk 2 above)
Comment (vmakarichev) [reply packages/SA/src/inputTools.ts:15]: Done

File: packages/SA/src/outputTools.ts (hunk 3 above)
Comment (vmakarichev) [reply packages/SA/src/outputTools.ts:17]: GREAT THANKS FOR CODE REVIEW!

**Review by AufarZakiev — COMMENTED** [review]
Will review later, after UI and computation polishing

**Pull request merged**